src/core/probe/kernel/kernel.rs

```diff
@@ -253,6 +265,27 @@ impl Kernel {
         Ok(())
     }
 
+    /// Request a hook to be attached to a specific target.
+    ///
+    /// ```
+    /// mod hook {
+    ///     include!("bpf/.out/hook.rs");
+    /// }
+    ///
+    /// [...]
+    ///
+    /// kernel.register_hook_to(hook::DATA, ProbeType::Kprobe, "kfree_skb_reason")?;
+    /// ```
+    pub(crate) fn register_hook_to(
```

Comment:
Why keep this API? The new one is much better :-) and completely supersedes this one.